core/utils/common/sqlite.go

```diff
@@ -0,0 +1,86 @@
+package common
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"time"
+
+	"github.com/glebarez/sqlite"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+)
+
+func LoadDBConnByPath(fullPath, dbName string) *gorm.DB {
+	if _, err := CreateDirWhenNotExist(true, path.Dir(fullPath)); err != nil {
+		panic(fmt.Errorf("init db dir failed, err: %v", err))
+	}
+	if _, err := os.Stat(fullPath); err != nil {
+		f, err := os.Create(fullPath)
+		if err != nil {
+			panic(fmt.Errorf("init %s db file failed, err: %v", dbName, err))
+		}
+		_ = f.Close()
+	}
+
+	db, err := GetDBWithPath(fullPath)
+	if err != nil {
+		panic(err)
+	}
+	return db
+}
+
+func LoadDBConnByPathWithErr(fullPath, dbName string) (*gorm.DB, error) {
+	if _, err := CreateDirWhenNotExist(true, path.Dir(fullPath)); err != nil {
+		return nil, fmt.Errorf("init db dir failed, err: %v", err)
+	}
+	if _, err := os.Stat(fullPath); err != nil {
+		f, err := os.Create(fullPath)
+		if err != nil {
+			return nil, fmt.Errorf("init %s db file failed, err: %v", dbName, err)
+		}
+		_ = f.Close()
+	}
+
+	db, err := GetDBWithPath(fullPath)
+	if err != nil {
+		return nil, fmt.Errorf("init %s db failed, err: %v", dbName, err)
+	}
+	return db, nil
+}
+
+func CloseDB(db *gorm.DB) {
+	sqlDB, err := db.DB()
+	if err != nil {
+		return
+	}
+	_ = sqlDB.Close()
+}
+
+func GetDBWithPath(dbPath string) (*gorm.DB, error) {
+	db, _ := gorm.Open(sqlite.Open(dbPath), &gorm.Config{
+		DisableForeignKeyConstraintWhenMigrating: true,
+		Logger:                                   newLogger(),
+	})
+	sqlDB, dbError := db.DB()
+	if dbError != nil {
+		return nil, dbError
+	}
+	sqlDB.SetConnMaxIdleTime(10)
+	sqlDB.SetMaxOpenConns(100)
+	sqlDB.SetConnMaxLifetime(time.Hour)
+	return db, nil
+}
+
+func newLogger() logger.Interface {
+	return logger.New(
+		log.New(os.Stdout, "\r\n", log.LstdFlags),
+		logger.Config{
+			SlowThreshold:             time.Second,
+			LogLevel:                  logger.Silent,
+			IgnoreRecordNotFoundError: true,
+			Colorful:                  false,
+		},
+	)
+}
```

Comment:
This code is missing important parts such as database connection configuration, session management logic etc. to fulfill its original function (LoadDBConnByPath). The `CreateDir` and `os.Stat` calls seem to be part of the creation process, but without further context this cannot be determined. There's also the assumption that the function will eventually close the GORM DB connection which is not specified.

To fix these issues:

1. Properly set up the database connection and configure logging.
   
2. Implement session management using packages like gorilla/mux if used on web services.

3. Remove unnecessary comments where no information has been provided.

4. Define specific return types or structure according to project needs to ensure proper implementation.

It would really assist understanding to have more details about the entire project setup and requirements; for example how many tables do you currently need? What kind of operations need it to handle?

5. Include an explicit check for empty paths or failure states during execution

Remember to run through a detailed review with all developers before proceeding because every environment could be different. 

I hope those suggestions helped! Please let me know if you want any clarification or help on other tasks.